src/semaphore_internal.h

```diff
@@ -90,4 +99,11 @@ _dispatch_put_thread_semaphore(_dispatch_thread_semaphore_t sema)
 	}
 }
 
+#if USE_FUTEX_SEM
+int _dispatch_futex_init(dispatch_futex_t *dfx);
```

Comment:
This should be

```C
#define DISPATCH_FUTEX_INIT  ((dispatch_futex_t){ 0 })
```

and no need for _dispatch_futex_dispose() I don't think it helps readability